sqlmesh/core/engine_adapter/base.py

```diff
@@ -425,61 +517,98 @@ def _create_table_from_source_queries(
+            and self.SUPPORTS_CTAS_SCHEMA_COMMENTS
+            and columns_to_types
+        ):
+            schema = self._build_schema_exp(table, columns_to_types, column_descriptions)
+
         with self.transaction(condition=len(source_queries) > 1):
             for i, source_query in enumerate(source_queries):
                 with source_query as query:
                     if i == 0:
                         self._create_table(
-                            table,
+                            schema if schema else table,
                             query,
                             columns_to_types=columns_to_types,
                             exists=exists,
                             replace=replace,
+                            table_description=table_description,
                             **kwargs,
                         )
                     else:
                         self._insert_append_query(
                             table_name, query, columns_to_types or self.columns(table)
                         )
 
+        # Register comments with commands if the engine doesn't support comments in the schema
+        if (
+            (table_description or column_descriptions)
+            and self.COMMENT_CREATION.is_comment_command_only
+        ) or (
+            column_descriptions
+            and self.COMMENT_CREATION.is_in_schema_def
+            and (not self.SUPPORTS_CTAS_SCHEMA_COMMENTS or not columns_to_types)
+        ):
+            self._create_comments(
+                table_name,
+                table_description,
+                column_descriptions,
+            )
+
     def _create_table(
         self,
         table_name_or_schema: t.Union[exp.Schema, TableName],
         expression: t.Optional[exp.Expression],
         exists: bool = True,
         replace: bool = False,
         columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+        table_description: t.Optional[str] = None,
+        column_descriptions: t.Optional[t.Dict[str, str]] = None,
         **kwargs: t.Any,
     ) -> None:
         self.execute(
-            self._create_table_exp(
+            self._build_create_table_exp(
                 table_name_or_schema,
                 expression=expression,
                 exists=exists,
                 replace=replace,
                 columns_to_types=columns_to_types,
+                table_description=table_description
+                if self.COMMENT_CREATION.is_in_schema_def
+                else None,
                 **kwargs,
             )
         )
 
-    def _create_table_exp(
+    def _build_create_table_exp(
         self,
         table_name_or_schema: t.Union[exp.Schema, TableName],
         expression: t.Optional[exp.Expression],
         exists: bool = True,
         replace: bool = False,
         columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+        table_description: t.Optional[str] = None,
```

Comment:
Seems like in case of this function, having a named `table_description` doesn't really contribute into the ease of use, so I might suggest keeping it in `kwargs`.